core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,260 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef KOKKOS_IMPL_PUBLIC_INCLUDE
+static_assert(false,
+              "Including non-public Kokkos header files is not allowed.");
+#endif
+
+#ifndef KOKKOS_BASIC_VIEW_HPP
+#define KOKKOS_BASIC_VIEW_HPP
+#include <type_traits>
+#include <Kokkos_Macros.hpp>
+#include <impl/Kokkos_Utilities.hpp>
+#include <impl/Kokkos_SharedAlloc.hpp>
+#include "Kokkos_ViewAlloc.hpp"
+#include "MDSpan/Kokkos_MDSpan_Accessor.hpp"
+#include "MDSpan/Kokkos_MDSpan_Header.hpp"
+
+namespace Kokkos {
+namespace Impl {
+template <class L>
+struct kokkos_layout_conversion {
+  using type = L;
+};
+
+template <>
+struct kokkos_layout_conversion<LayoutLeft> {
+  using type = Experimental::layout_left_padded<dynamic_extent>;
+};
+
+template <>
+struct kokkos_layout_conversion<LayoutRight> {
+  using type = Experimental::layout_right_padded<dynamic_extent>;
+};
```

Comment:
I am not convinced that this is a good idea. In the end `layout_type` should be what you hand as a template argument to `View` and I think we should have the nested `mdspan` have the same layout as the original thing.